### PR TITLE
[Inductor] Corner case empty graph

### PR DESCRIPTION
**Pull request by anijain2305**
Fixes the latest errors seen in https://github.com/pytorch/torchdynamo/issues/1148

### CODE REVIEW

File: torchinductor/compile_fx.py [hunk 1]
```diff
@@ -72,6 +73,9 @@ def compile_fx_inner(
     num_fixed=0,
     is_backward=False,
 ):
+    if count_calls(gm.graph) == 0:
```
Comment (voznesenskym) [inline torchinductor/compile_fx.py:76]: surprised we get here - iirc dynamo raises `raise exc.SkipFrame()` if `count_calls` returns 0. Did you see this downstream of dynamo?

Comment (anijain2305) [reply torchinductor/compile_fx.py:76]: Yeah, I am surprised too. This is DDP magic. Need to look into more details whats going on.

Comment (voznesenskym) [reply torchinductor/compile_fx.py:76]: Can you dump the graph? Perhaps its a graph of constants? Or something funky like an empty gm?

Comment (jansel) [reply torchinductor/compile_fx.py:76]: Could be from a backwards graph...

**Review by jansel — APPROVED** [review]

**Comment by voznesenskym** [comment]
> @voznesenskym I added the offending graph as a test. The original graph is not empty, but the AOT Autograd generated fwd graph is empty.

lovely, ty

**Review by voznesenskym — APPROVED** [review]

**Comment by anijain2305** [comment]
CI failure is unrelated.

**Pull request merged**